openupgrade_scripts/scripts/hr_fleet/16.0.1.0/pre-migration.py

```diff
@@ -0,0 +1,35 @@
+from openupgradelib import openupgrade
+
+
+def create_drive_employee(env):
+    openupgrade.logged_query(
+        env.cr,
+        """
+        ALTER TABLE fleet_vehicle_assignation_log
+        ADD driver_employee_id INT
+        """,
+    )
+
+    openupgrade.logged_query(
+        env.cr,
+        """
+        SELECT id, driver_employee_id
+        FROM fleet_vehicle
+        """,
+    )
+
+    for vehicle in env.cr.fetchall():
+        openupgrade.logged_query(
+            env.cr,
+            """
+            UPDATE fleet_vehicle_assignation_log
+            SET driver_employee_id = %s
+            WHERE vehicle_id = %s
+            """,
+            (vehicle[1], vehicle[0]),
+        )
+
+
+@openupgrade.migrate()
+def migrate(env, version):
+    create_drive_employee(env)
```

Comment:
```suggestion
@openupgrade.migrate()
def migrate(env, version):
    openupgrade.logged_query(
        env.cr,
        """
            ALTER TABLE fleet_vehicle_assignation_log
            ADD COLUMN IF NOT EXISTS driver_employee_id integer
        """,
    )
    openupgrade.logged_query(
        env.cr,
        """
        UPDATE fleet_vehicle_assignation_log AS fval
        SET driver_employee_id = v.driver_employee_id
        FROM fleet_vehicle AS v
        WHERE v.id = fval.vehicle_id
        """,
    )
```

Reply:
please keep a dedicated function to populate driver_employee_id. the migrate function should only contains call to others function.

Reply:
> please keep a dedicated function to populate driver_employee_id. the migrate function should only contains call to others function.

@legalsylvain Ok. I got it. Thank you